### PR TITLE
fix: use alternative data structure for dict ops

### PR DESCRIPTION
**Pull request by pgautier404**
This commit introduces an `Element` type that contains an `ElemField` and `ElemValue`, both of which use our `Value` type. This type is for use with `DictionarySetFields` and allows users to provide byte and string fields and values, as opposed to go's built-in `map` which disallows byte slice keys. It also includes helper functions to convert from a few different types of maps to `[]Element`. Finally, the shared test context code has been modified to prepend test caches with `golang-` for easy identification of the source of stranded caches.

### CODE REVIEW

**Review by cprice404 — DISMISSED** [review]
i think i like this direction, personally!  we could also provide a utility function that converts from map[string]Value to []Element.

**Review by poppoerika — COMMENTED** [review]
I like this approach as well! 
Also I don't think we need much education is need for Go users since I'd assume they're used to a struct param.

But since this is a breaking change, we need to include in 1.0 release though?

**Comment by pgautier404** [comment]
> But since this is a breaking change, we need to include in 1.0 release though?

Yes, we're definitely going to want to get this in for 1.0.

File: momento/value.go [hunk 1]
```diff
@@ -30,3 +30,7 @@ func (v String) asBytes() []byte {
 func (v String) asString() string {
 	return string(v)
 }
+
+type Element struct {
+	ElemField, ElemValue Value
+}
```
Comment (poppoerika) [inline momento/value.go:36]: Maybe add doc here to describe what `Element` is?
Something like `Element` is a Momento custom map type which accepts flexible types for map key and value or something like that?

File: momento/requester.go [hunk 2]
```diff
@@ -235,3 +233,36 @@ func validateNotEmpty(field []byte, label string) error {
 	}
 	return nil
 }
+
+func ElementsFromMapStringString(theMap map[string]string) []Element {
```
Comment (poppoerika) [inline momento/requester.go:237]: Are these functions customer facing so that users can easily create an `[]Element` value?

Comment (poppoerika) [reply momento/requester.go:237]: These means `ElementsFromMapStringString`, `ElementsFromMapStringBytes`, and `ElementsFromMapStringValue`

Comment (pgautier404) [reply momento/requester.go:237]: Yes, exactly.

File: momento/value.go (hunk 1 above)
Comment (pgautier404) [reply momento/value.go:36]: Good call, I'll add some docs.

Comment (pgautier404) [reply momento/value.go]: Fixed in 036fedb, thanks for the suggestion!

File: momento/requester.go (hunk 2 above)
Comment (poppoerika) [reply momento/requester.go:237]: Awesome. 🍾

File: momento/value.go (hunk 1 above)
Comment (poppoerika) [reply momento/value.go]: Thank you!

**Review by poppoerika — APPROVED** [review]
Looks good to me!
Thank you for doing this, Pete.

**Pull request merged**